.github/workflows/maintainer_bors.yml

```diff
@@ -0,0 +1,93 @@
+    # both set simultaneously: depending on the event that triggers the PR, usually only one is set
+    env:
+      AUTHOR: ${{ github.event.comment.user.login }}${{ github.event.review.user.login }}
+      PR_NUMBER: ${{ github.event.issue.number }}${{ github.event.pull_request.number }}
+      COMMENT_EVENT: ${{ github.event.comment.body }}
+      COMMENT_REVIEW: ${{ github.event.review.body }}
+      PR_TITLE_ISSUE: ${{ github.event.issue.title }}
+      PR_TITLE_PR: ${{ github.event.pull_request.title }}
+      PR_URL: ${{ github.event.issue.html_url }}${{ github.event.pull_request.html_url }}
+      EVENT_NAME: ${{ github.event_name }}
+    name: Add ready-to-merge or delegated label
+    runs-on: ubuntu-latest
+    steps:
+      - name: Find bors merge/delegate
+        id: merge_or_delegate
+        run: |
+          COMMENT="${COMMENT_EVENT}${COMMENT_REVIEW}"
+          # we strip `\r` since line endings from GitHub contain this character
+          COMMENT="${COMMENT//$'\r'/}"
+          # for debugging, we print some information
+          printf '%s' "${COMMENT}" | hexdump -cC
+          printf 'Comment:"%s"\n' "${COMMENT}"
+          m_or_d="$(printf '%s' "${COMMENT}" |
+            sed -n 's=^bors  *\(merge\|r+\)$=ready-to-merge=p; s=^bors  *d.*=delegated=p' | head -1)"
+
+          printf $'"bors delegate" or "bors merge" found? \'%s\'\n' "${m_or_d}"
+
+          printf $'mOrD=%s\n' "${m_or_d}" > "${GITHUB_OUTPUT}"
+          if [ "${AUTHOR}" == 'leanprover-community-mathlib4-bot' ] ||
+             [ "${AUTHOR}" == 'leanprover-community-bot-assistant' ]
+          then
+            printf $'bot=true\n'
+            printf $'bot=true\n' > "${GITHUB_OUTPUT}"
+          else
+            printf $'bot=false\n'
+            printf $'bot=false\n' > "${GITHUB_OUTPUT}"
+          fi
+
+      - name: Check whether user is a mathlib admin
+        id: user_permission
+        if: ${{ ! steps.merge_or_delegate.outputs.mOrD == '' }}
+        uses: actions-cool/check-user-permission@v2
+        with:
+          require: 'admin'
+          # review(_comment) use
+          # require: 'write'
+          # token: ${{ secrets.TRIAGE_TOKEN }}
+
+      - name: Add ready-to-merge or delegated label
+        id: add_label
+        if: ${{ ! steps.merge_or_delegate.outputs.mOrD == '' &&
+          ( steps.user_permission.outputs.require-result == 'true' ||
+            steps.merge_or_delegate.outputs.bot == 'true' ) }}
+        uses: octokit/request-action@v2.x
+        with:
+          # check is this ok? was /repos/:repository/issues/:issue_number/labels
+          route: POST /repos/:repository/issues/${PR_NUMBER}/labels
+          repository: ${{ github.repository }}
+          issue_number: ${PR_NUMBER}
+          labels: '["${{ steps.merge_or_delegate.outputs.mOrD }}"]'
+        env:
+          GITHUB_TOKEN: ${{ secrets.TRIAGE_TOKEN }} # comment uses ${{ secrets.GITHUB_TOKEN }}
```

Comment:
What does this comment mean?

Reply:
This comment is highlighting a difference between the merged actions: the action triggered by a comment would use `GITHUB_TOKEN`, while the actions triggered by `review` or `review_comment` would use `TRIAGE_TOKEN`.  As a consequence of the Zulip discussion, it seemed better to use `TRIAGE` to alleviate the GH quota.